### PR TITLE
Minor Rules Engine and pipeline improvements

### PR DESCRIPTION
**Pull request by tonybach**
JIRAs:
- https://issues.cask.co/browse/CDAP-12374
- https://issues.cask.co/browse/CDAP-12409

### CODE REVIEW

**Comment by tonybach** [comment]
Also added a commit for https://issues.cask.co/browse/CDAP-11885

File: cdap-ui/app/cdap/components/DataPrep/Directives/ExtractFields/index.js [hunk 1]
```diff
@@ -52,19 +55,34 @@ export default class ExtractFields extends Component {
...
+        })}>
           <div
             onClick={this.parseUsingPatterns}
             className="option"
           >
             {T.translate(`${PREFIX}.patternSubmenuTitle`)}
           </div>
         </div>
+        {
+          this.isUsingPatternsDisabled? (
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/DataPrep/Directives/ExtractFields/index.js]: nit: space before `?`

File: cdap-ui/app/cdap/components/DataPrep/Directives/ExtractFields/index.js [hunk 2]
```diff
@@ -52,19 +55,34 @@ export default class ExtractFields extends Component {
...
 
+    let usingPatternsId = 'extract-fields-using-patterns';
+
     return (
       <div
         className="extract-fields second-level-popover"
         onClick={this.preventPropagation}
       >
-        <div className="extract-field-options">
+        <div id={usingPatternsId}
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/DataPrep/Directives/ExtractFields/index.js]: Have all the attributes to `div` in a newline. 

```
<div 
  id="..."
  className={...}
>
  ...
</div>
```

File: cdap-ui/app/cdap/components/RulesEngineHome/RuleBookDetails/RuleBookDetails.scss [hunk 3]
```diff
@@ -20,6 +20,7 @@
 $loading_bar_color: white;
 $rule-book-metadata-color: #999999;
 $header-bg-color: #f5f5f5;
+$rule-book-create-fields-required-color: #d40001;
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/RulesEngineHome/RuleBookDetails/RuleBookDetails.scss]: We already have brand-danger `d15668` is this something that is common for all the `required fields` label? @leacd We should either use the existing red or we should start using this as common color. Right now there are too many reds and maroons that are being used in different places.

**Review by ajainarayanan — COMMENTED** [review]
Minor comments.

**Comment by ajainarayanan** [comment]
@tonybach did we change the Create rule book design to have Created on the right? Didn't see this in the mocks attached with the JIRA. Just checking.
screen shot 2017-09-05 at 1 37 18 pm

**Comment by tonybach** [comment]
@ajainarayanan Hmm that's because of the `display: flex` and `justify-content: space-between` added in this commit: https://github.com/caskdata/cdap/commit/c4ab63e2f7d7943659b4dbda61893518bb8ea635#diff-d7bf85f30822bc05c73e365abe694976. Not sure if we need that or not?

File: cdap-ui/app/cdap/components/RulesEngineHome/RuleBookDetails/RuleBookDetails.scss (hunk 3 above)
Comment (tonybach) [reply cdap-ui/app/cdap/components/RulesEngineHome/RuleBookDetails/RuleBookDetails.scss]: Talked with @leacd, we want to keep both colors. We can use the existing `brand-danger` for error backgrounds, while this new color will be for error text. I extracted this out to `variables.scss` and called it `cdap-red-text`. Feel free to suggest a better variable name.

**Comment by ajainarayanan** [comment]
Hhmm based on the mocks its not something we would want. Lets restore it to one below the other. Not sure how I missed it.

**Review by ajainarayanan — COMMENTED** [review]
👍 Code LGTM

**Comment by leacd** [comment]
LGTM

**Pull request merged**